public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,885 @@
+  They even have their own modules: the `ReactiveFormsModule` and the `FormsModule`.
+
+  ### _Reactive_ forms
+  Angular _reactive_ forms facilitate a _reactive style_ of programming 
+  that favors explicit management of the data flowing between
+  a non-UI _data model_ (typically retrieved from a server) and a 
+  UI-oriented _form model_ that retains the states 
+  and values of the HTML controls on screen.
+
+  With _reactive_ forms, you create a tree of Angular form controls in code, and 
+  bind them to HTML widgets in the component template, using techniques described in this guide.
+
+  The component class is responsible for all data flowing between the component and the screen.
+  It has immediate access to both the data model and the form control structure. 
+  It pushes data model values into the form controls and 
+  pulls user-changed values back out of the form controls.
+  The component can _observe_ changes in form control state and _react_ to those changes.
+
+  In keeping with the reactive paradigm, the component should preserve the immutability of the _data model_,
+  treating it as a pure source of original values.
+  Rather than update the data model directly, 
+  the component extracts user changes and forwards them to an external component or service,
+  which does something with them (such as saving them) 
+  and returns a new _data model_ to the component that reflects the updated model state.
+
+  ### _Template-driven_ forms
+
+  _Template-driven_ forms, introduced in the [Template guide](forms.html), take a completely different approach.
+  
+  You place HTML form controls (such as `<input>` and `<select>`) in the component template and 
+  bind them to _data model_ properties in the component, using the `[(ngModel)]` two-way data binding directive.
+  
+  You don't create Angular form controls. Angular infers them from the data bindings.
+  You don't push and pull data values. Angular handles that for you with `[(ngModel)]`.
+  Angular updates the mutable _data model_ with user changes as they happen.
+
+  Both data binding and the mutable _data model_ are contrary to reactive principles.
+  In fact, the `ngModel` directive is only available in the `FormsModule`, not in the `ReactiveFormsModule`.
+
+  These diffences don't make _template-driven forms_ wrong and _reactive forms_ right.
+  These are different architectural paradigms, with their own strengths and weaknesses,
+  and you are free to choose between them.
+
+  The balance of this _reactive forms_ guide assumes the _reactive_ paradigm and 
+  concentrates exclusively on the reactive forms technology and techniques. For 
+  information on _template driven forms_, see the [Template Guide](forms.html).
+
+  You'll learn about reactive forms by building one from scratch,
+  with instances of the Angular form classes.
+
+  ### Essential form classes
+
+  Before you begin, it may be helpful to read a brief description of the core form classes.
```

Comment:
I see your point, since the doc is long. This explanation being in the beginning of the videos really helped me get my bearings. I like them in the beginning for that reason but am not married to it. What are your thoughts @wardbell?